### PR TITLE
Update/status cell design

### PR DESCRIPTION
**Pull request by vykes-mac**
This PR adopt the status column design to match the new design.
image

## Proposed Changes

* Redesign status cell
* Count the number of alerts requiring attention and display in the header

## Testing Instructions

* go to `/domains/manage?flags=domains/management`
* Verify the design matches p9Jlb4-8DU-p2 
* Verify the correct number of alerts is displayed in the header, it should match what's in prod

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/domains-table/src/domains-table-header/index.tsx [hunk 1]
```diff
@@ -104,6 +106,9 @@ export const DomainsTableHeader = ( {
 							tabIndex={ column?.isSortable ? 0 : -1 }
 						>
 							{ column?.headerComponent || column?.label }
+							{ column?.name === 'status' && domainsRequiringAttention && (
+								<span className="list-status-cell__bubble">{ domainsRequiringAttention }</span>
+							) }
```
Comment (p-jackson) [inline packages/domains-table/src/domains-table-header/index.tsx:111]: Would we be able to hide this somehow while the rows are still loading? It seems a little peculiar having a number here while the column itself is still flashing placeholders.

CleanShot 2023-08-30 at 21 25 01@2x

File: packages/domains-table/src/status-popover/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	const handleOnMouseLeave = () => {
+		setTimeout( () => {
+			if ( inPopover ) {
+				return;
+			}
+
+			setShowPopover( false );
+		}, 250 );
+	};
```
Comment (p-jackson) [inline packages/domains-table/src/status-popover/index.tsx]: I discovered that sometimes the popover would be stuck open. I'm not sure exactly how I did it with the mouse, but I guess it's a timing thing. Somehow `inPopover` behind left `true` when the timeout expires.

I tried to see if I could get a better result using the `useDebounce` hook (from the `use-debounce` package). I think I had better results by just tracking the hover state of each element and then debouncing the value of `inPopover || inButton`.

```tsx
	const [ inPopover, setInPopover ] = useState( false );
	const [ inButton, setInButton ] = useState( false );
	const [ showPopover ] = useDebounce( inPopover || inButton, 250 );

	const handleOnMouseEnterButton = () => {
		setInButton( true );
	};

	const handleOnMouseLeave = () => {
		setInButton( false );
	};

	const handleOnMouseEnterPopover = () => {
		setInPopover( true );
	};

	const handleOnMouseLeavePopover = () => {
		setInPopover( false );
	};
```

**Review by p-jackson — APPROVED** [review]
Looks good!

I find it a bit confusing that the rows do a calculation and then call a callback in the domains table. But I think it's ok because if we end up replacing `/all-domains` with a different endpoint, then that new endpoint could maybe return us the total count of rows that need attention too.

I would have merged but I didn't want to mess up the rebase, and there are a lot of commits there that I didn't want to do something wrong with and break 😄

**Comment by vykes-mac** [comment]
> I find it a bit confusing that the rows do a calculation and then call a callback in the domains table. But I think it's ok because if we end up replacing /all-domains with a different endpoint, then that new endpoint could maybe return us the total count of rows that need attention too.

We could do the calculations in the domains-table `index.tsx` file but we still would need some indication that the domain details have finish loading in the `DomainTableRow` so the header can show the number in errors or its gonna always show the number while the rows are still loading as it is now.

**Comment by vykes-mac** [comment]
Close in favour of https://github.com/Automattic/wp-calypso/pull/81231 since the history got messed up in this one.

**Comment by p-jackson** [comment]
> We could do the calculations in the domains-table index.tsx file but we still would need some indication that the domain details have finish loading in the DomainTableRow

Oh I see now. Yeah the row does need that data too. I feel like this is something React Query can help us with. Like, if both components are using the same query then they will both have access to the query's data and `isLoading` data, but only one request gets sent over the network. We don't have to limit the React Query hook to only being in one component and then passing that data around as a prop, or using a callback function. At least that's how I feel we should be able to do it, but maybe it's more complicated than that.

@vykes-mac

**Pull request closed without merging**